File: client/lib/features/teacher/domain/entities/teacher.dart
import 'package:equatable/equatable.dart';

import '../../data/model/teacher_model.dart';

class Teacher extends Equatable {
  final String id;
  final String username;
  final String email;
  final String? subject;
  final bool isHomeroom;
  final bool isDirector;
  final List<SchoolClass>? classes;

  const Teacher({
    required this.id,
    required this.username,
    required this.email,
    this.subject,
    this.isHomeroom = false,
    this.isDirector = false,
    this.classes,
  });

  @override
  List<Object?> get props => [
    id,
    username,
    email,
    subject,
    isHomeroom,
    isDirector,
    classes,
  ];
}

File: server/app/users/schemas/teacher.py
import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, EmailStr

from app.users.schemas.user_base import UserRead
from app.classes.schemas import ClassSlim


class TeacherBase(BaseModel):
    user_id: uuid.UUID
    subject: Optional[str]
    is_director: Optional[bool] = False
    is_homeroom: Optional[bool] = False
    class_id: Optional[uuid.UUID] = None
    school_id: Optional[uuid.UUID] = None

    class Config:
        from_attributes = True


class TeacherCreate(TeacherBase):
    username: str
    email: EmailStr


class TeacherUpdate(TeacherCreate):
    pass


class TeacherRead(TeacherBase):
    user: UserRead
    classes: list[ClassSlim] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
